internal/cmd/initcmd.go

```diff
@@ -133,9 +133,22 @@ func (c *Config) runInitCmd(cmd *cobra.Command, args []string) error {
 		c.init.purgeBinary = true
 	}
 
-	// If the source repo does not exist then init or clone it.
-	switch _, err := c.baseSystem.Stat(c.SourceDirAbsPath.Join(".git")); {
-	case errors.Is(err, fs.ErrNotExist):
+	// Find the working copy.
+	workingCopyDirAbsPath := c.SourceDirAbsPath
+FOR:
+	for workingCopyDirAbsPath != "/" {
```

Comment:
This reminds me of https://nodejs.org/api/modules.html#modules_loading_from_node_modules_folders. It should work!

I would just have delegated the work to `git` instead, as it is already a dependency of chezmoi anyways.

Reply:
chezmoi actually doesn't have a hard dependency on git, see the [`--use-builtin-git` flag](https://github.com/twpayne/chezmoi/blob/master/docs/REFERENCE.md#--use-builtin-git-value).

Reply:
Hm... what I understand by `--user-builtin-git` is that chezmoi ships a git binary inside of it. This mean git is always available, even when not installed in the computer. No?

Reply:
It embeds [github.com/go-git/go-git](https://github.com/go-git/go-git), which is an implementation of git in a Go library. It means that limited git functionality (`init` and `update`) is available, even if git is not installed on the machine. The main reason for adding this was to avoid requiring the user to install git on ephemeral machines before being able to install their dotfiles.

Reply:
Oh okay! Thank you for the explanation! Now I understand.